contracts/WtapETH.sol

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.18;
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Permit.sol";
+import "./interfaces/ITapETH.sol";
+
+/**
+ * @title TapETH token wrapper with static balances.
+ * @dev It's an ERC20 token that represents the account's share of the total
+ * supply of tapETH tokens. WtapETH token's balance only changes on transfers,
+ * unlike tapETH that is also changed when staking rewards and swap fee are generated.
+ * It's a "power user" token for DeFi protocols which don't
+ * support rebasable tokens.
+ * The contract is also a trustless wrapper that accepts tapETH tokens and mints
+ * wtapETH in return. Then the user unwraps, the contract burns user's wtapETH
+ * and sends user locked tapETH in return.
+ * The contract provides the staking shortcut: user can send ETH with regular
+ * transfer and get wtapETH in return. The contract will send ETH to Tapio
+ * staking it and wrapping the received tapETH.
+ *
+ */
+
+contract WtapETH is ERC20Permit {
+    ITapETH public tapETH;
+
+    constructor(
+        ITapETH _tapETH
+    ) ERC20Permit("Wrapped tapETH") ERC20("Wrapped tapETH", "wtapETH") {
+        tapETH = _tapETH;
+    }
+
+    /**
+     * @notice Exchanges tapETH to wtapETH
+     * @param _tapETHAmount amount of tapETH to wrap in exchange for wtapETH
+     * @dev Requirements:
+     *  - msg.sender must approve at least `_tapETHAmount` tapETH to this
+     *    contract.
+     * @return Amount of wtapETH user receives after wrap
+     */
+    function wrap(uint256 _tapETHAmount) external returns (uint256) {
+        require(_tapETHAmount > 0, "wtapETH: can't wrap zero tapETH");
+        uint256 _wtapETHAmount = tapETH.getSharesByPooledEth(_tapETHAmount);
+        _mint(msg.sender, _wtapETHAmount);
+        tapETH.transferFrom(msg.sender, address(this), _tapETHAmount);
```

Comment:
```suggestion
        tapETH.transferFrom(msg.sender, address(this), _tapETHAmount);
        _mint(msg.sender, _wtapETHAmount);
```

Reply:
in term of security pattern , we should have 
        _mint(msg.sender, _wtapETHAmount);
        tapETH.transferFrom(msg.sender, address(this), _tapETHAmount);